Sources/Controllers/Map/Layers/OAPOILayer.mm

```diff
@@ -436,6 +457,34 @@ - (OATargetPoint *) getTargetPoint:(id)obj
         targetPoint.sortIndex = (NSInteger)targetPoint.type;
         return targetPoint;
     }
+    else
+    {
+        OATargetPoint *targetPoint = [[OATargetPoint alloc] init];
+        targetPoint.type = OATargetLocation;
+        targetPoint.location = CLLocationCoordinate2DMake(renderedObject.labelLatLon.latitude, renderedObject.labelLatLon.longitude);
+        targetPoint.values = renderedObject.tags;
+        targetPoint.obfId = renderedObject.obfId;
+        targetPoint.targetObj = renderedObject;
+        targetPoint.sortIndex = (NSInteger)targetPoint.type;
+        
+        if (!poi)
+            poi = [RenderedObjectViewController getSyntheticAmenityWithRenderedObject:renderedObject];
```

Comment:
Move such static methods from view controller to an helper or other common class